specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Dataset.json

```diff
@@ -282,7 +282,7 @@
     "CompressionLevel": {
       "title": "Microsoft.Azure.Management.DataFactories.Models.CompressionLevel",
       "description": "All available compression levels.",
-      "type": "object",
```

Comment:
I don't think string is going to work for datafactory. This is meant to be oneOf scenario and swagger 2.0 hasn't supported oneOf yet.

Reply:
cc @lirenhe

Reply:
It has to be string as `"modelAsString": true"` below.

If it is not string, modelAsString should be false, and `format` need to be provided.

Reply:
Not that, In datafactory, this kind of property can be either enum string or Expression type. we need to use oneOf to express such scenario which is only supported in swagger 3.0, but currently we are using swagger 2.0 and I believe, we don't have plan to upgrade to swagger 3.0 at least in recent months.

Reply:
I think I get you. But even if this is supported by swagger, it will not be supported by SDK, at least not Java. We will not return anything like such (i.e. it could be enum, it could be string otherwise not included in the enum, or it could be an object of undefined schema, as there is no `Expression` anywhere in this def).
Hence Java will not release for datafactory.